### ISSUE TITLE
Issues with quivers/arrows

### ISSUE THREAD

**Issue by AgentOrange1719** [userofbricks/Expanded-Combat#20]
I've run into several issues related to quivers and arrows, not sure if they're related or not. I created a new clean install to test these issues without other mods running, and I still encounter them. Maybe I'm just doing something wrong, but I'd love to use this mod if I can get it working for me. 

Minecraft 1.17.1
Forge 37.0.119

Mods:
Curios API (Forge) 1.17.1-5.0.1.0
Expanded Combat 2.3.1-beta_2-1.17.1
Sophisticated Backpacks 1.17.1-3.11.1.394 (To test curio slots and see if Curio API was the issue, seems not to be as issues persisted in a new install w/o SB)

Issues:
1. The game crashes to desktop when removing a quiver from the quiver curio slot. This happens whether or not there are any arrows in the quiver, and also with any quiver type. Notably, this seems to somehow break Curios API. When I relaunch the game, I cannot open any curio inventory. The button to show the curios lots appears, but clicking it does nothing, nor does using the hotkey to open the curio slots. I can only fix this by disabling Curios API and launching the game, loading in, quitting, and reenabling Curios API. Removing items from other curio slots works fine, including this mod's gauntlets, backpacks from Sophisticated Backpacks, etc.
2. A possibly related issue I discovered while testing the first one is that the game also crashes to desktop when a mob dies after being shot with an arrow fired from a quiver. This does not happen if there are no arrows in a quiver, or no quiver is equipped, and the arrows are therefor being drawn from the standard inventory. This also does not happen when a mob is shot with an arrow from a quiver, only when the shot kills it, and the type of arrow does not matter.
3. I've also noticed that killing a mob with an arrow fired from the curio slot causes the same crash as when using an arrow from a quiver.
4.  The quiver HUD does not show up, though using the hotkey to open the quiver GUI works. The hotkeys to cycle arrow types does not work, and will always fire the first stack.
5. 
I recorded a video showing the entire process of creating the fresh install with CurseForge, installing Curios API and this mod, and showing the issues. It's a little over 11 minutes, though that's with a lot of loading time. I'll link it here as an unlisted YouTube video. https://youtu.be/Kcd_NWbGlaA Maybe I'm just doing something stupid. I've also attached all three crash reports from the CTSs shown in the video

[crash-2021-11-25_15.39.08-server.txt](https://github.com/userofbricks/Expanded-Combat/files/7605601/crash-2021-11-25_15.39.08-server.txt)
[crash-2021-11-25_15.40.43-server.txt](https://github.com/userofbricks/Expanded-Combat/files/7605602/crash-2021-11-25_15.40.43-server.txt)
[crash-2021-11-25_15.45.10-server.txt](https://github.com/userofbricks/Expanded-Combat/files/7605603/crash-2021-11-25_15.45.10-server.txt)

**Comment by userofbricks** [comment]
man. yeah that versions is bound to have more bugs than just these but that is more because of how i implemented things when updating from 1.16.5 some of these issues will hopfully get solved when I change certain systems to work a bit safer than ther current iterations. the fourth issue is known and I am still trying to wrap my head around why. hopefully irl work will slow down for me soon so I can work on this mod without feeling absolutely exausted.

**Comment by Coraline77447** [comment]
I'm having the same issues on forge 37.1.0 but no crashes but from reading the report above now I'm a little bit afraid that my game will crash if I try to mess with the quivers and arrows so now I'll have to get rid of the first quiver that I'd made while restricting myself to only use the flint arrows to avoid the game to crash but not to desktop since I have not enough crashes installed. 
Also what I've noticed is that whenever I'm tryin to equip the leather quiver with my flint arrows that my 60 flint arrows have mysteriously vanished after hitting their re-binded keybindings from their defaults since they were conflicting with my other keybindings.
I think that the flint arrows are in the quiver but which one tho since I have two of them but I only made one? (I was in creative mode when I attempted to add my flint arrows to the leather quiver that I'd crafted in creative mode but I dont know if that has something to do with it tho)...
Also I cant access the Curios inventorys anymore, even hitting g does nothing, so now I'm suspecting that I can't equip the gauntlets of any type or anything else in Expanded Combat anymore...
Even tho I want to keep playing with Explanded Comabt but I'm going to uninstall and wait till you come around to fix these issues....
